File: Backend/src/main/java/tn/esprit/kaddem/entities/Projet.java
package tn.esprit.kaddem.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;
import java.util.Date;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

public class Projet implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long idProjet;
    String description;
    double prixProjet;


    @ManyToOne
    Equipe equipe;

    @ManyToOne
    Entreprise entrepriseP;

    private LocalDate dateDebutProjet;
    private LocalDate dateFinProjet;
    @Enumerated(EnumType.STRING)
    private Status status;
    @Enumerated(EnumType.STRING)
    private Review review;


}

File: FrontEnd/src/app/core/model/projet.ts
import {Entreprise} from "./entreprise";
import {Equipes} from "./equipes";

export class Projet {
  idProjet: number;
  description: String;
  prixProjet:number;
  review: String;
  equipe:Equipes;
  entrepriseP: Entreprise;
  status:String;
  dateDebutProjet:Date;
  dateFinProjet:Date;
}

